### ISSUE TITLE
Stuck on blank page login in...

### ISSUE THREAD

**Issue by rems25000** [9p4/jellyfin-plugin-sso#113]
**Describe the bug**
Some user gets stuck on "login in..."

**To Reproduce**
Correctly identified on the authentic side, it is redirected to Jellyfin but remains blocked on the white page.

**Expected behavior**
This is related to the account in question, no problem with the others. The user still has a non-deactivated Jellyfin listed account.

**Workaround**

1.  In "SSO-Auth.xml" I remove the block linked to the user.

```
 <item>
              <key>
                <string>username@yahoo.com</string>
              </key>
              <value>
                <guid>cac8167e-1daf-476a-9af7-568e7495214e</guid>
              </value>
            </item>
```

2.  I restart Jellyfin.
**3.  the user has normal access again.**

**observation**
All users who have had the problem tell me that they used a mobile client before having the problem, check if it is related or not.

**Comment by 9p4** [comment]
Can you please upload the Jellyfin logs

**Comment by rems25000** [comment]
**log**
[2023-03-13 12:13:31.336 +01:00] [INF] SSO Controller initialized
[2023-03-13 12:13:32.771 +01:00] [INF] SSO Controller initialized
[2023-03-13 12:13:34.215 +01:00] [INF] Is request linking: False
[2023-03-13 12:13:34.648 +01:00] [INF] SSO Controller initialized
[2023-03-13 12:13:34.667 +01:00] [INF] Auth request created...
[2023-03-13 12:13:34.667 +01:00] [INF] Current/Max sessions for user "userhere@gmail.com": 1/1
[2023-03-13 12:13:34.669 +01:00] [ERR] Error processing request: "User is at their maximum number of sessions". URL "POST" "/sso/OID/Auth/oauth".

Effectively, users are entitled to a single session. But when you "disconnect" normally, he is entitled to a new session, and there, I don't see him connected.

**Comment by 9p4** [comment]
does this issue arise if the users sign out first?

**Comment by rems25000** [comment]
I managed to reproduce the problem with one of my accounts.
Condition 1, the right account has only one session. 
I normally connect from a computer browser.
I disconnect and reconnect several times correctly from Jellyfin, Always -> connection... -> Connected OK!
I disconnect one last time from Jellyfin and switch to a mobile browser (for my example safari IOS).
authentication OK -> I am redirected to Login in... and the page remains blocked.
From my point of view, when I logged in on mobile, I see the account "in session" on Jellyfin but it blocks on the blank login page.
The account can’t log in, can’t log out in Jellyfin’s eyes. 
log:
[2023-03-14 00:56:11.838 +01:00] [INF] SSO Controller initialized
[2023-03-14 00:56:13.172 +01:00] [INF] SSO Controller initialized
[2023-03-14 00:56:14.636 +01:00] [INF] Is request linking: True
[2023-03-14 00:56:15.227 +01:00] [INF] SSO Controller initialized
[2023-03-14 00:56:15.245 +01:00] [INF] Auth request created...
[2023-03-14 00:56:15.245 +01:00] [INF] Current/Max sessions for user "test@gmail.com": 1/1
[2023-03-14 00:56:15.249 +01:00] [ERR] Error processing request: "User is at their maximum number of sessions". URL "POST" "/sso/OID/Auth/oauth".

**Comment by 9p4** [comment]
What version of Jellyfin and the plugin are you using?

**Comment by rems25000** [comment]
Jellyfin Version : 10.8.9
SSO-Auth 3.4.0.0

**Comment by rems25000** [comment]
![image](https://user-images.githubusercontent.com/97801362/225711528-241d86d0-1a44-4d20-995e-677e6659dab5.png)
No version beyond 3.4.0.0 from the repository:
https://raw.githubusercontent.com/9p4/jellyfin-plugin-sso/manifest-release/manifest.json

**Comment by 9p4** [comment]
That's weird, there should be a 3.4.0.2. Anyway, are you willing to edit the source code of the plugin to try and figure out where the issue is coming from?

**Comment by rems25000** [comment]
In any case, I can tell you that it only affects people who connect from an iPhone.

**Comment by 9p4** [comment]
https://github.com/9p4/jellyfin-plugin-sso/blob/c6cac45004a32a7d8cf5863102ef947f792ed021/SSO-Auth/Api/SSOController.cs#L643

Can you add `user.MaxActiveSessions = 100;`? Then, check the logs to see if the sessions are always increasing or do they cap out somewhere.

**Comment by rems25000** [comment]
I can't find any "SSOController.cs" files on my file system, here are the known locations:
/usr/share/jellyfin/web
/var/lib/jellyfin/plugins/configurations
/var/lib/jellyfin/plugins/SSO Authentication_3.4.0.0:
![image](https://user-images.githubusercontent.com/97801362/225722890-6e1be88d-bc33-4eb1-99f3-36cb817df1b8.png)

**Comment by 9p4** [comment]
Yes, you will need to download the plugin source and compile it from scratch.

Or I could compile it for you and send you the `.dll` if you want.

**Comment by rems25000** [comment]
Yes, I want the modified DLL :)

**Comment by 9p4** [comment]
[SSO-Auth.dll.zip](https://github.com/9p4/jellyfin-plugin-sso/files/10994846/SSO-Auth.dll.zip)

**Comment by rems25000** [comment]
I replaced the file
Restart Services
I managed to log in once on iPhone
I logged out.
I tried to reconnect -> white page login in...

in log:
[2023-03-16 19:59:54.247 +01:00] [INF] SSO Controller initialized
[2023-03-16 19:59:54.267 +01:00] [INF] Auth request created...
[2023-03-16 19:59:54.267 +01:00] [INF] Current/Max sessions for user "test@gmail.com": 0/1
[2023-03-16 19:59:54.269 +01:00] [INF] Creating new access token for user cb6145d7-f1c6-46d2-81bc-xxxxxxxxxxxx
[2023-03-16 19:59:55.670 +01:00] [INF] "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2023-03-16 19:59:55.670 +01:00] [INF] "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2023-03-16 19:59:55.675 +01:00] [INF] AuthenticationScheme: "CustomAuthentication" was challenged.
[2023-03-16 19:59:56.481 +01:00] [INF] "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2023-03-16 19:59:56.481 +01:00] [INF] "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2023-03-16 19:59:56.481 +01:00] [INF] AuthenticationScheme: "CustomAuthentication" was challenged.


[2023-03-16 20:00:30.777 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:00:31.757 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:00:33.265 +01:00] [INF] Is request linking: False
[2023-03-16 20:01:10.103 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:01:10.121 +01:00] [INF] Auth request created...
[2023-03-16 20:01:10.121 +01:00] [INF] Current/Max sessions for user "test@gmail.com": 1/1
[2023-03-16 20:01:10.124 +01:00] [ERR] Error processing request: "User is at their maximum number of sessions". URL "POST" "/sso/OID/Auth/oauth".

**Comment by 9p4** [comment]
Can you delete the user and remove the link

**Comment by rems25000** [comment]
I am no longer stuck on the blank page.
I manage to log in once.
If I disconnect and disconnect I am no longer stuck on the white page, but I return each time to the authentication link, it goes in a loop.

LOG:

[2023-03-16 20:20:45.235 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:20:46.435 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:20:48.034 +01:00] [INF] Is request linking: False
[2023-03-16 20:20:48.291 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:20:48.299 +01:00] [INF] SSO user test@gmail.com doesn't exist, creating...
[2023-03-16 20:20:48.593 +01:00] [INF] SSO user link doesn't exist, creating...
[2023-03-16 20:20:48.619 +01:00] [INF] Auth request created...
[2023-03-16 20:20:48.622 +01:00] [INF] Current/Max sessions for user "test@gmail.com": 0/100
[2023-03-16 20:20:48.665 +01:00] [INF] Creating new access token for user 6a13f2fe-60f7-4406-8c53-55bf6add4af9
[2023-03-16 20:20:48.712 +01:00] [WRN] User "test@gmail.com" was found with invalid/missing Authentication Provider "Jellyfin.Plugin.SSO_Auth.Api.SSOController". Assigning user to InvalidAuthProvider until this is corrected
[2023-03-16 20:20:49.264 +01:00] [WRN] User "test@gmail.com" was found with invalid/missing Authentication Provider "Jellyfin.Plugin.SSO_Auth.Api.SSOController". Assigning user to InvalidAuthProvider until this is corrected

[2023-03-16 20:21:03.977 +01:00] [INF] Logging out access token "ccdc2e3add7448118ce193cf35dcc6ab"

[2023-03-16 20:21:14.727 +01:00] [INF] Auth request created...
[2023-03-16 20:21:14.727 +01:00] [INF] Current/Max sessions for user "test@gmail.com": 0/100
[2023-03-16 20:21:14.729 +01:00] [INF] Creating new access token for user 6a13f2fe-60f7-4406-8c53-55bf6add4af9
[2023-03-16 20:21:14.754 +01:00] [WRN] User "test@gmail.com" was found with invalid/missing Authentication Provider "Jellyfin.Plugin.SSO_Auth.Api.SSOController". Assigning user to InvalidAuthProvider until this is corrected
[2023-03-16 20:21:15.112 +01:00] [INF] "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2023-03-16 20:21:15.112 +01:00] [INF] "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2023-03-16 20:21:15.117 +01:00] [INF] AuthenticationScheme: "CustomAuthentication" was challenged.
[2023-03-16 20:21:20.111 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:21:21.017 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:21:22.368 +01:00] [INF] Is request linking: False
[2023-03-16 20:21:22.587 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:21:22.614 +01:00] [INF] Auth request created...
[2023-03-16 20:21:22.614 +01:00] [INF] Current/Max sessions for user "test@gmail.com": 1/100
[2023-03-16 20:21:22.617 +01:00] [INF] Reissuing access token: "cb7a8d38bb6e4790aa9b7e660b7132e1"
[2023-03-16 20:21:22.617 +01:00] [WRN] User "test@gmail.com" was found with invalid/missing Authentication Provider "Jellyfin.Plugin.SSO_Auth.Api.SSOController". Assigning user to InvalidAuthProvider until this is corrected
[2023-03-16 20:21:29.638 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:21:30.727 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:21:31.937 +01:00] [INF] Is request linking: False
[2023-03-16 20:21:32.218 +01:00] [INF] SSO Controller initialized
[2023-03-16 20:21:32.240 +01:00] [INF] Auth request created...
[2023-03-16 20:21:32.240 +01:00] [INF] Current/Max sessions for user "test@gmail.com": 1/100
[2023-03-16 20:21:32.243 +01:00] [INF] Reissuing access token: "cb7a8d38bb6e4790aa9b7e660b7132e1"
[2023-03-16 20:21:32.243 +01:00] [WRN] User "test@gmail.com" was found with invalid/missing Authentication Provider "Jellyfin.Plugin.SSO_Auth.Api.SSOController". Assigning user to InvalidAuthProvider until this is corrected

**Comment by 9p4** [comment]
What part specifically is looping?

**Comment by rems25000** [comment]
On iPhone only, the login page is displayed for 1 sec and instead of being written on Jellyfin home, I return to the Jellyfin login page.
On PC its working perfectly fine.

**Comment by 9p4** [comment]
Strange. I'll see if I can reproduce your issue.

**Comment by rems25000** [comment]
OK, I managed to reproduce on my PC browser simply with user agent example:Safari/604.1"
"Mozilla/5.0 (iPad; CPU OS 11_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.0 Tablet/15E148 Safari/604.1"
![image](https://user-images.githubusercontent.com/97801362/225894252-980107db-4494-4c3f-8da6-f9cd55e0af3e.png)

**Comment by 9p4** [comment]
Does quick connect work?

**Comment by rems25000** [comment]
This feature is not enabled at the moment, I only restrict to SSO.

**Comment by 9p4** [comment]
Could you enable it? It's the recommended way to use clients that don't have access to a web browser.

**Comment by 9p4** [comment]
Unfortunately, I can't reproduce your issue.

**Comment by rems25000** [comment]
With two sessions allowed per user, I haven't had any other complaints yet.

**Comment by JaySmithWpg** [comment]
I managed to accidentally recreate this issue tonight. My Nginx configuration originally contained this location directive, as per [this guide](https://jellyfin.org/docs/general/networking/nginx/):
```
    location = / {
        return 302 http://$host/web/;
        #return 302 https://$host/web/;
    }
```
This resulted in me getting stuck on the "Logging in..." page, as the browser blocks the insecure http redirect after the SSO page (mixed content).

Updating my config to the following fixed the issue:
```
    location = / {
        return 302 https://$host/web/;
    }
```
I hope this helps! Hopefully you have the same issue and aren't facing a different bug with the same results.

**Comment by 9p4** [comment]
@JaySmithWpg did you also have the following in your nginx config?

```
location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Protocol $scheme;
        proxy_set_header X-Forwarded-Host $http_host;
}
```

**Comment by rems25000** [comment]
Good morning,
I thought the problem was solved, but I just saw that I have an avalanche of requests "stuck on a blank page"
nothing to do with nginx always forced to delete the user and the <CanonicalLinks> to unlock the user.
Please find the problem, it becomes critical.

**Comment by 9p4** [comment]
@rems25000 can you upload your nginx configuration?

**Comment by 9p4** [comment]
I can't reproduce your error. I am using the unmodified plugin on multiple iOS devices across multiple users.

**Comment by rems25000** [comment]
With a mac that I don't have.

**Comment by rems25000** [comment]
Ideally, it would be to have an error message in the page that says why the page stays on "login in..." and hangs during the process.

**Comment by 9p4** [comment]
What OpenID are you using?

**Comment by rems25000** [comment]
authentik

**Comment by rems25000** [comment]
Hello,
What does each user's guid correspond to in the config file?
Because if it hooks an identifier linked to the browser, namely, IOS uses random identifiers against tracking in its privacy policy.
I managed to reproduce the problem with an Android, the person just cleared their browser cache and couldn't reconnect afterwards.

**Comment by 9p4** [comment]
It links the SSO provider's UID that it responds with.

**Comment by 9p4** [comment]
I have yet to reproduce your issue :(

**Comment by rems25000** [comment]
I can no longer connect to my personal account and I find it in the logs.

[2023-04-03 22:18:03.251 +02:00] [INF] SSO Controller initialized
[2023-04-03 22:18:05.007 +02:00] [INF] Is request linking: False
[2023-04-03 22:18:05.380 +02:00] [INF] SSO Controller initialized
[2023-04-03 22:18:05.395 +02:00] [ERR] Error processing request. URL "POST" "/sso/OID/Auth/oauth".
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.GetOrCreateEntry(Object entity, IEntityType entityType)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.EntryWithoutDetectChanges(TEntity entity)
   at Microsoft.EntityFrameworkCore.Internal.InternalDbSet`1.Update(TEntity entity)
   at Jellyfin.Server.Implementations.Users.UserManager.UpdateUserAsync(User user)
   at Jellyfin.Server.Implementations.Users.UserManager.UpdateUserAsync(User user)
   at Jellyfin.Plugin.SSO_Auth.Api.SSOController.Authenticate(Guid userId, Boolean isAdmin, Boolean enableAuthorization, Boolean enableAllFolders, String[] enabledFolders, AuthResponse authResponse, String defaultProvider)
   at Jellyfin.Plugin.SSO_Auth.Api.SSOController.OidAuth(String provider, AuthResponse response)
   at lambda_method1145(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context)

**Comment by 9p4** [comment]
Can you try to set the default authentication provider to `Jellyfin.Server.Implementations.Users.DefaultAuthenticationProvider`?

**Comment by rems25000** [comment]
Already done, without result, and jellyfin support says it's the plugin that damaged the db.
when I try to watch a video it cuts all the time and we can clearly see an error related to the plugin.

```
[2023-04-08 22:49:47.564 +02:00] [ERR] [21] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL "POST" "/Sessions/Playing/Progress".
Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while saving the entity changes. See the inner exception for details.
 ---> Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'NOT NULL constraint failed: Users.AuthenticationProviderId'.
   at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
   at Microsoft.Data.Sqlite.SqliteDataReader.NextResult()
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(StateManager stateManager, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Jellyfin.Server.Implementations.Users.UserManager.UpdateUserAsync(User user)
   at Jellyfin.Server.Implementations.Users.UserManager.UpdateUserAsync(User user)
   at Emby.Server.Implementations.Session.SessionManager.LogSessionActivity(String appName, String appVersion, String deviceId, String deviceName, String remoteEndPoint, User user)
   at Jellyfin.Api.Helpers.RequestHelpers.GetSession(ISessionManager sessionManager, IAuthorizationContext authContext, HttpRequest request)
   at Jellyfin.Api.Helpers.RequestHelpers.GetSessionId(ISessionManager sessionManager, IAuthorizationContext authContext, HttpRequest request)
   at Jellyfin.Api.Controllers.PlaystateController.ReportPlaybackProgress(PlaybackProgressInfo playbackProgressInfo)
   at lambda_method1260(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager)
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context)

```

**Comment by 9p4** [comment]
Is this still an issue?

**Comment by Thesola10** [comment]
Okay, this may be a CORS issue due to the OIDC provider being on another domain, and my nginx reverse proxy config added two `Access-Control-Allow-Origin` headers due to shenanigans. Will keep posted as soon as I find a working CORS.

**Comment by Thesola10** [comment]
Okay, I must say, this has to be the most puzzling OIDC handshake ever. Somehow the "Logging in..." page is itself going back and pinging the callback endpoint?? Like, now the failing CORS request is `/login?callback_uri=/apps/oidc/redirect`, which should NEVER be a CORS request in the first place?? No wonder the whole thing's positively broken.

**Comment by ItzGigglez** [comment]
I'm currently working through an issue getting this working with an Authentik/Traefik/Docker setup and ran into several of these symptoms along the way, and so far I've only been able to get it working with version 3.4.0.0.

I noticed that when I ever have one of my configurations that gets stuck at the "Logging in..." page, there appears to be a logged in version of the Jellyfin App embedded in an iframe with width and height set to 0... removing the styling seems to "let you in" so to speak. Its seems to be behaving like there's a script that's supposed to make the iframe 'full screen' and it isn't getting triggered?

**Comment by 9p4** [comment]
The iframe is there to set the cookies. The size of the iframe shouldn't matter. Could you do a git bisect and try to narrow down the commit that introduced the issue?

**Comment by ItzGigglez** [comment]
I can give it a shot, but I'm still actually quite new to using git so I'm not entirely sure how to do that. 

Just to clarify, doing the bisect would entail cloning this repo, using bisect to mark 3.4.0.0 as 'good' and 3.5.0.0 as 'bad', then stepping through the commits recompiling from source? 

Also, I've been installing the plugin through Jellyfins GUI, so I'm assuming to test I would need to then copy the compiled files somewhere into Jellyfins filesystem?

**Comment by ItzGigglez** [comment]
I just tried stepping through the versions available through the Jellyfin GUI and I'm now running into a different series of issues that seem to be somewhat related, but are not technically the same 'stuck at a screen that says "Logging in..."' issue as the original, I'm going to play around with my config to see if I can recreate this exact error. Not sure if I should create a new issue for my new error state, or post the logs here? (TLDR: `error processing request url get /sso/OID/r/authentik, endpoint belongs to different authority`)

While I have a 'working' configuration with the 3.4.0.0 version of the plugin, I'm not entirely sure how much of the issue (in my case at least, for both the "Logging in..." error, and the error state with my current config) is actually the plugin, vs a domain level forward-auth I have setup in authentik prior to the Jellyfin login screen.

**Comment by 9p4** [comment]
Could you try setting the `doNotValidateEndpoints` option to `true`?

**Comment by 9p4** [comment]
(this was a defaults change in 3.5 to make the plugin more secure)

**Comment by 9p4** [comment]
Enabling this options usually means that something is badly misconfigured on the OIDC provider end, but some providers (like Google) have a weird architecture that necessitates this option.

**Comment by ItzGigglez** [comment]
Enabling `doNotValidateEndpoints` in 3.5.2.0 does seem to have resolved the `endpoint belongs to different authority` issue.
